app/src/main/java/com/asfoundation/wallet/ui/iab/PaymentMethodsPresenter.kt

```diff
@@ -432,6 +434,13 @@ class PaymentMethodsPresenter(
     }
       .andThen(
         getWalletInfoUseCase(null, cached = false)
+          .doOnSuccess {
+            activity?.let { activity ->
+              ChallengeRewardManager.create(
```

Comment:
presenter is not ideal for it to be responsible for opening an activity, you should raise this value to the Fragment to then call the activity

Reply:
Yes, i forgot about an extra step there, it might be what you're talking about